_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,324 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that
+
+- has the [semantic role][] of [rowheader][] or [columnheader][] and;
+- is [visible][] and;
+- is [included in the accessibility tree][] and;
+- whose closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][];
+  1. has [descendants][descendant], that are non empty ("") and;
```

Comment:
How would it be possible that the `table` ancestor has only empty descendant?
Can you build an example where this is the only part of the Applicability that is not matched?